nx/lib/nx/defn/grad.ex

```diff
@@ -507,6 +507,62 @@ defmodule Nx.Defn.Grad do
     to_grad(t, g, cache)
   end
 
+  defp grad(:take, [t, i, axis], _ans, g, cache) do
+    axes_range = 0..(tuple_size(t.shape) - 1)//1
```

Comment:
```suggestion
    axes_range = 0..(tuple_size(t.shape) - 1)//1
```

Do we use `Nx.rank(t)` or `tuple_size(t.shape)` in this file? make sure we are following one "standard" or the other. :)

Reply:
Nx.rank :)

I'll change this and possibly other occurences